logger.js

```diff
@@ -1,13 +1,8 @@
 const AWS = require('aws-sdk');
 
 class Logger {
-    constructor({stage, stack, app, roleArn, streamName, sendLogsToKinesis = true}) {
-        this.stage = stage;
-        this.stack = stack;
-        this.app = app;
-        this.roleArn = roleArn;
-        this.streamName = streamName;
-        this.sendLogsToKinesis = sendLogsToKinesis;
+    constructor(options) {
+        Object.assign(this, { sendLogsToKinesis: true }, options)
```

Comment:
This does look a **lot** nicer! Does jsdoc/flow types enforce required args or is it more a hint?

We could do something like this to enforce required args? Its pretty verbose though 😕 

```js
const required = ['I', 'Am', 'Required'];
const provided = new Set([...Object.keys(options)]);

const isOptionsValid = new Set([...required].filter(r => provided.has(r)).size === required.length;

if (! isOptionsValid) {
  throw 'missing options'
}
else {
  // continue
}
```